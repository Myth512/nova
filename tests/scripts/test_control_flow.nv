x := 1

// Case 1, expected: true
if x == 1 {
	print(true)
}

// Case 1.5, expected: nothing
if x != 1 {
	print(true)
}

// Case 2, expected: true
if x > 0 {
	print(true)
} else {
	print(false)
}

// Case 3, expected: false
if x < 0 {
	print(true)
} else {
	print(false)
}

// Case 4, expected: true
if x > 0 {
	print(true)
} elif x == 0 {
	print("Maybe")
} else {
	print(false)
}

// Case 5, expected: Maybe
x = 0
if x > 0 {
	print(true)
} elif x == 0 {
	print("Maybe")
} else {
	print(false)
}

// Case 6, expected: false
x = -1

if x > 0 {
	print(true)
} elif x == 0 {
	print("Maybe")
} else {
	print(false)
}

// Case 7, expected: false
a, b := false, false

if a == true and b == true {
	print(true)
} else {
	print(false)
}

// Case 8, expected: false
a, b = false, true

if a == true and b == true {
	print(true)
} else {
	print(false)
}

// Case 9, expected: false
a, b = true, false

if a == true and b == true {
	print(true)
} else {
	print(false)
}
	
// Case 10, expected: true
a, b = true, true 

if a == true and b == true {
	print(true)
} else {
	print(false)
}

// Case 11, expected: false
a, b = false, false

if a == true or b == true {
	print(true)
} else {
	print(false)
}

// Case 12, expected: true
a, b = false, true

if a == true or b == true {
	print(true)
} else {
	print(false)
}

// Case 13, expected: true
a, b = true, false

if a == true or b == true {
	print(true)
} else {
	print(false)
}

// Case 14, expected: true
a, b = true, true

if a == true or b == true {
	print(true)
} else {
	print(false)
}

def dangerousFunction() {
	print("Im very dangerous")
}

// Case 15, expected: false
if false and dangerousFunction() {
	print(true)
} else {
	print(false)
}

// Case 16, expected: true
if true or dangerousFunction() {
	print(true)
} else {
	print(false)
}
