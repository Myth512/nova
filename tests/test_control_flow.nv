x := 1

if x == 1 {
	print("1: True")
}

if x != 1 {
	print("2: True")
}

if x > 0 {
	print("3: True")
} else {
	print("3: False")
}

if x < 0 {
	print("4: True")
} else {
	print("4: False")
}

if x > 0 {
	print("5: True")
} elif x == 0 {
	print("5: Maybe")
} else {
	print("5: False")
}

x = 0

if x > 0 {
	print("6: True")
} elif x == 0 {
	print("6: Maybe")
} else {
	print("6: False")
}

x = -1

if x > 0 {
	print("7: True")
} elif x == 0 {
	print("7: Maybe")
} else {
	print("7: False")
}

a, b := false, false

if a == true and b == true {
	print("8: True")
} else {
	print("8: False")
}

a, b = false, true

if a == true and b == true {
	print("9: True")
} else {
	print("9: False")
}

a, b = true, false

if a == true and b == true {
	print("10: True")
} else {
	print("10: False")
}
	
a, b = true, true 

if a == true and b == true {
	print("11: True")
} else {
	print("11: False")
}

a, b = false, false

if a == true or b == true {
	print("12: True")
} else {
	print("12: False")
}

a, b = false, true

if a == true or b == true {
	print("13: True")
} else {
	print("13: False")
}

a, b = true, false

if a == true or b == true {
	print("14: True")
} else {
	print("14: False")
}

a, b = true, true

if a == true or b == true {
	print("15: True")
} else {
	print("15: False")
}

def dangerousFunction() {
	print("Im very dangerous")
}

if false and dangerousFunction() {
	print("16: True")
} else {
	print("16: False")
}

if true or dangerousFunction() {
	print("17: True")
} else {
	print("17: False")
}
