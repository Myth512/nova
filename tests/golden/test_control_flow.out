true
true
false
true
Maybe
false
false
false
false
true
false
true
true
true
false
true
